### PR TITLE
feat: added support for open source models, added instructions to docs

### PR DESCRIPTION
**Pull request by ErikBjare**
Hi @AntonOsika,

Thought I'd give the new WizardCoder models a go, they are pretty impressive!

I was going to try and run the benchmarks, but it was a bit finicky, so I did this instead :)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -74,6 +74,16 @@ Editing the `preprompts`, and evolving how you write the project prompt, is how
 
 Each step in `steps.py` will have its communication history with GPT4 stored in the logs folder, and can be rerun with `scripts/rerun_edited_message_logs.py`.
 
+### Running with open source models
```
Comment (ErikBjare) [inline README.md:77]: This section might go better in the docs (which I just found), although concerned the instructions will simply be buried there. Maybe we could just do it, throw in a link from the README, and call it a day.

File: README.md [hunk 2]
```diff
@@ -74,6 +74,16 @@ Editing the `preprompts`, and evolving how you write the project prompt, is how
...
+
+You can use gpt-engineer with open source models by using an OpenAI compatible API, such as the one offered by the [text-generator-ui extension `openai`](https://github.com/oobabooga/text-generation-webui/blob/main/extensions/openai/README.md). This can easily be setup with [TheBloke's Runpod template](https://www.runpod.io/console/gpu-secure-cloud?template=f1pf20op0z).
+
+To do so, first set up the API according to the instructions linked above.
```
Comment (ErikBjare) [inline README.md:81]:
```suggestion
To do so, first set up the API according to the instructions linked above. Then you need to go into the text-generation-webui, go to settings, check the `openai` extension, save. You then need to expose TCP port 5001 in your Runpod config, which will give it an exposed TCP port something like 40125. Then restart your Runpod, and check that the API is live by browsing: http://<public ip>:<port>/v1/models
```

**Comment by AntonOsika** [comment]
Thanks @ErikBjare! 🏃 

Regarding README, it's already too long (want to keep it super short) so I think your idea of linking to docs is the way to go!

Would you be able to update PR?

**Comment by ErikBjare** [comment]
Done! Moved it into the docs, and also added some link to the docs from the README (there wasn't a single one! :scream:)

**Comment by ErikBjare** [comment]
Oops, pushed commits to wrong remote so this PR didn't have all the changes. Addressing in #644

**Pull request merged**